### ISSUE TITLE
[Feature Request] r-base image tags for each minor release

### ISSUE THREAD

**Issue by averikitsch** [rocker-org/rocker#349]
Currently, r-base only has tags for complete releases. This means as soon as you release a new image, this will be out of date. 

Ideally, there would be an immutable tag for each release, eg `3.6.0`, and a tag that's updated with every patch image release, eg `3.6`.

**Comment by eddelbuettel** [comment]
I am sorry but I don't understand what you are asking for.  Do you want a 'two-digit' tag ie 3.6 from 3.6.0, rolled to .3.6.1, then to a potential 3.6.2, in addition to the existing tags?

I just grep here. `sort` and `tail` help as well.  My current random snapshot on my box:

```sh
edd@rob:~$ docker images | grep ^r-base
r-base                       3.6.1               4e55790c88ae        4 weeks ago         642MB
r-base                       latest              4e55790c88ae        4 weeks ago         642MB
r-base                       3.6.0               6dfc0dec6d66        3 months ago        662MB
r-base                       3.5.3               62c848eeb175        4 months ago        649MB
r-base                       3.5.2               578644a0540c        7 months ago        650MB
r-base                       3.5.1               93ca38345db2        13 months ago       678MB
r-base                       3.5.0               fb43e8fe1667        13 months ago       698MB
r-base                       3.4.3               d1e1c25485af        17 months ago       670MB
r-base                       3.4.2               02d3b7e00020        21 months ago       651MB
r-base                       3.4.0               5a6c58403310        2 years ago         656MB
r-base                       3.3.3               88436550cddc        2 years ago         635MB
r-base                       3.3.1               7ba1baf9d8bb        2 years ago         657MB
edd@rob:~$ 
```

(Docker's r-base is the same as our rocker/r-base; I keep it as it is easier to type.)

**Comment by averikitsch** [comment]
Right, every patch release,`3.6.X`, would be rolling to a tag `3.6`, as well as the specific tag, `3.6.X`. 
This is a common pattern for Docker tags (https://hub.docker.com/_/alpine?tab=tags).
This allows users to stay up to date on patches without the risk of a breaking change in the next minor release.

**Comment by eddelbuettel** [comment]
I'll try to remember to do this next time.  BTW I disagree with your premise: _This means as soon as they release a new image, this will be out of date._  Not so.  When I rolled up 3.6.1 and tagged it as 3.6.1 it ensure it _will forever be accessible under that tag_ which is the purpose.  It will always give us 3.6.1.  You have a different question: "what is the most recent R release".  You should really ask the R Project about that (and can).   But we could probably do this tagging as well ...

**Comment by cboettig** [comment]
Yeah, we do have post-build hooks that create tags for the minor versions
on the r-ver side, eg r-ver:3.5 is identical to r-ver:3.5.3, see
https://hub.docker.com/r/rocker/r-ver/tags

Haven’t checked these recently though.


-- 
---
Carl Boettiger
http://carlboettiger.info/

**Comment by eddelbuettel** [comment]
Different build process than r-base though.

**Comment by averikitsch** [comment]
@eddelbuettel It is great that it will be forever accessible under tag `3.6.1`, but if I want the current patch I would have to go look it up. Having a tag `3.6`, would allow me to keep up to date on the patches without wasting my time to go look up the new release and updating my Dockerfile every time there is a new patch. I also can't use the tag `latest`, because I am worried about my build breaking with major/minor changes. This doesn't change the functionality of having pinned versions but allows for a better developer experience due to the flexibility. 

TLDR; as a user I want to stay up to date on new releases, but I don't want to worry about breaking my build.

**Comment by eddelbuettel** [comment]
Not to beat a dead horse but you also have to "roll" from 3.4 to 3.5 to 3.6 to ... Hence rolling between actual picked release (which differ between them) is important for me too.  But hey, each at their own.  I'll to fold this in next time.

**Comment by eitsupi** [comment]
`rocker/r-ver` (R >= 4.0.0) and derived images, built from [rocker-org/rocker-versioned2](https://github.com/rocker-org/rocker-versioned2), now push major versioned tag (`rocker/r-ver:4`) and minor versioned tags (`rocker/r-ver:4.0` and `rocker/r-ver:4.1`).
Please check the wiki attached to the repository for a list of tags.
https://github.com/rocker-org/rocker-versioned2/wiki

**Comment by eddelbuettel** [comment]
@eitsupi This question is for a completely differrent stack and answere here:

- for Rocker: 
  https://hub.docker.com/repository/docker/rocker/r-base          

- for the official R image derived from it:    
  https://hub.docker.com/_/r-base

Both have had proper tags for years (though not from the beginning)

For a discussion of the different stacks in Rocker, see the R Journal paper:  https://journal.r-project.org/archive/2017/RJ-2017-065/index.html

**Comment by eitsupi** [comment]
@eddelbuettel Of course I understand that, I posted it here because https://github.com/rocker-org/rocker-versioned/issues/211 was closed as a duplicate of this issue.

**Comment by eddelbuettel** [comment]
This is a different repo. It covers a different product.

**Comment by eddelbuettel** [comment]
Closing this because the issue was, in fact, addressed years ago -- and I expressed regret that I didn't do it earlier.  

Tags are good.  Docker Hub is bad in only supporting a fixed number of them.  But _that_ is a different issue.

**Comment by eitsupi** [comment]
I apologize for posting in the inappropriate place.

**Comment by eddelbuettel** [comment]
You meant well -- and I should have closed this earlier. All good.
